### PR TITLE
Add resource Dataproc batch

### PR DESCRIPTION
**Pull request by zli82016**
fixes https://github.com/hashicorp/terraform-provider-google/issues/11775

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_dataproc_batch`
```

```release-note:enhancement
dataproc: switched to the v1 API for `google_dataproc_autoscaling_policy` resource (beta)
```

### CODE REVIEW

**Comment by zli82016** [comment]
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_dataproc_batch` (5 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```hcl
> resource "google_dataproc_batch" "primary" {
>   runtime_config {
>     container_image = # value needed
>   }
>   spark_batch {
>     archive_uris      = # value needed
>     file_uris         = # value needed
>     main_jar_file_uri = # value needed
>   }
>   spark_r_batch {
>     archive_uris = # value needed
>     file_uris    = # value needed
>   }
> }
> ```

Hello, @melinath, I added the test for most fields. There are still a couple of fields missing the test. I talked with the service team, and they may need about 2 days to set up. They want to check if we are OK to go without those fields in the test.

File: mmv1/products/dataproc/product.yaml
Comment (melinath) [inline mmv1/products/dataproc/product.yaml:14]: Why is this temporary?

**Comment by melinath** [comment]
What are they setting up? Could we just leave those fields out of the resource for now?

**Comment by zli82016** [comment]
> What are they setting up? Could we just leave those fields out of the resource for now?

They need to create a public GAR image, write spark jobs for these fields and create respective files and archives. The setup is just for testing.

File: mmv1/products/dataproc/product.yaml
Comment (zli82016) [reply mmv1/products/dataproc/product.yaml]: The comment should not be here. I removed it.

**Comment by melinath** [comment]
If they can get it set up before merge that would definitely be preferable, but I think we can go ahead otherwise if the overall resource works. There's just a higher risk of bugs.

**Review by melinath — APPROVED** [review]
LGTM - Adding the tests would be great if it works out before the next cut, otherwise it's probably fine to go ahead with this.

**Comment by melinath** [comment]
Actually it would be great to merge this Monday if possible.

**Comment by zli82016** [comment]
I am going to merge this PR now and add the missing test later.

**Pull request merged**
